jerry-core/ecma/base/ecma-globals.h

```diff
@@ -690,15 +690,31 @@ typedef enum
 } ecma_string_container_t;
 
 /**
+ * ECMA string related constants
+ */
+typedef enum
+{
+  ECMA_STRING_CONTAINER_MASK = 0x7u, /**< container mask */
+  ECMA_STRING_REF_ONE = 1u << 3, /**< value for increasing or decreasing the reference counter */
+  ECMA_STRING_MAX_REF = 0x1fffu << 3, /**< maximum value of the reference counter (8191) */
+} ecma_string_constants_t;
```

Comment:
You didn't see them? I didn't like it either, but this is the preferred way in Jerry. Just search for "typedef enum" and you can find a lot of similar ones. E.g jerry_flag_t, jerry_fatal_code_t, vm_oc_get_types, ecma_property_flags_t, re_flags_t. The reason was something like you cannot misuse them. The compiler gives better error messages for them than for defines.

I will add a static assert.